#59 - Add @StreamedCache annotation (#62)
Iteo/cached: packages/cached/README.md

- if we have too many `clearAllCached` annotation, only one can be
- if method don’t return `bool`, `Future<bool>` or not a `void`

### Streamed cache

> Method annotation that is used to get a stream of cache updates from a cached method.
> You have to provide at least the name of the cached class method in the `methodName` parameter.

Reviewer: ```suggestion
Use `cachedMethod` annotation to get a stream of cache updates from a cached method. Remember to provide at least the name of the cached class method in the `methodName` parameter.
```

Simple example of usage:
```dart
@cached
int cachedMethod() {
  return 1;  
}

@StreamedCache(methodName: "cachedMethod", emitLastValue: true)
Stream<int> cachedStream();
```

Parameters of cachedMethod should match target method (except `@ignore` or `@ignoreCache`), otherwise `InvalidGenerationSourceError` will be thrown.

> Return type of method marked with `@StreamedCache` should be `Stream<sync type of target method>`.

Reviewer: This is a bit complex to understand, I think we could paraphrase it

- if we have too many `clearAllCached` annotation, only one can be
- if method don’t return `bool`, `Future<bool>` or not a `void`

### Streamed cache

Method annotation that is used to get a stream of cache updates from a cached method.
You have to provide at least the name of the cached class method in the `methodName` parameter.

Simple example of usage:
```dart
@cached
int cachedMethod() {
  return 1;  
}

@StreamedCache(methodName: "cachedMethod", emitLastValue: true)
Stream<int> cachedStream();
```

Parameters of cachedMethod should match target method (except `@ignore` or `@ignoreCache`), otherwise `InvalidGenerationSourceError` will be thrown.
Return type of method marked with `@StreamedCache` should be `Stream<sync type of target method>`.

Example:
```dart
@cached
Future<String> cachedMethod(int x, @ignore String y) async {
  await Future.delayed(Duration(miliseconds: 100));
  return x.toString();  
}

@StreamedCache(methodName: "cachedMethod", emitLastValue: false)
Stream<String> cachedStream(int x);
```

> You can also request usage of `BehaviorSubject` from RxDart instead of `StreamController.broadcast()` in generated code, by setting `useBehaviorSubject` to true.

Reviewer 2: We shouldn't use `BehaviorSubject`. `emitLastValue: true` should handled without any external dependency
